GearRent/Networking/APIManager.h

```diff
@@ -6,14 +6,18 @@
 //
 
 #import <Foundation/Foundation.h>
-#import "Item.h"
+#import "Listing.h"
 #import "GNGeoHash.h"
+#import "Category.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
 @interface APIManager : NSObject
 
-void fetchListingsWithCoordinates(NSArray<CLLocation *> *coordinates, void(^completion)(NSArray<Item *> *, NSError *error));
+void fetchListingsWithCoordinates(NSArray<CLLocation *> *coordinates, void(^completion)(NSArray<Listing *> *, NSError *error));
+- (void)fetchNearestCity:(CLLocation *)location completion: (void(^_Nonnull)(NSString *, NSError *)) completion;
+void fetchAllCategories(void(^completion)(NSArray<Category *> *, NSError *));
+void fetchDynamicPrice(Listing *listing, void(^completion)(float, NSError *));
```

Comment:
for all the C functions, you can move them out of the `@interface APIManager` definition, you might need to add `extern` before the return type

Reply:
@eninodel  mostly for styling but C function might be a bit faster I'd guess.